support/util/util.go

```diff
@@ -458,3 +458,109 @@ func PredicatesForHostedClusterChildResourcesAnnotationScoping(r client.Reader)
 	}
 	return predicate.NewPredicateFuncs(filter)
 }
+
+// PredicatesForNodepoolAnnotationScoping returns predicate filters for all event types that will ignore incoming
+// event requests for nodepool resources in which their owning hostedcluster resource doesn't have a scope annotation
+// that matches what is specified in the HOSTEDCLUSTERS_SCOPE_ANNOTATION env var.  If not defined or empty, the default behavior
+// is to accept all nodepool events in which the owning hostedcluster resource does not have a corresponding scope annotation defined.
+// The ENABLE_HOSTEDCLUSTERS_ANNOTATION_SCOPING env var must also be set to "true" to enable the scoping feature.
+func PredicatesForNodepoolAnnotationScoping(r client.Reader) predicate.Predicate {
+	hcAnnotationScopingEnabledEnvVal := os.Getenv(EnableHostedClustersAnnotationScopingEnv)
+	hcScopeAnnotationEnvVal := os.Getenv(HostedClustersScopeAnnotationEnv)
+	filter := func(obj client.Object) bool {
+		if hcAnnotationScopingEnabledEnvVal != "true" {
+			return true
+		}
+
+		np, ok := obj.(*hyperv1.NodePool)
+		if !ok {
+			return true
+		}
+
+		// use the Cluster Name from the nodepool spec to get the owning hostedcluster object
+		hostedClusterName := ""
+		if np.Spec.ClusterName != "" {
+			hostedClusterName = np.Spec.ClusterName
+		}
+		if hostedClusterName == "" {
+			return true
+		}
+		namespacedName := ParseNamespacedName(fmt.Sprintf("%s/%s", np.Namespace, hostedClusterName))
```

Comment:
You can just create a `types.NamespacedName{}` literal using the fields - no need to string-format and parse.